lib/ui/painting.dart

```diff
@@ -2855,11 +2858,49 @@ class ImageFilter {
       : assert(matrix4 != null), // ignore: unnecessary_null_comparison
         _data = Float64List.fromList(matrix4),
         _filterQuality = filterQuality,
+        _colorFilter = null,
+        _innerFilter = null,
+        _outerFilter = null,
         _type = _kTypeMatrix {
     if (matrix4.length != 16)
       throw ArgumentError('"matrix4" must have 16 entries.');
   }
 
+  /// Creates an image filter from a [ColorFilter].
+  ///
+  /// The resulting filter is functionally equivalent to the original
+  /// [ColorFilter], but can be used and composed like an [ImageFilter]. For
+  /// example, the [BackdropFilter] widget applies an [ImageFilter] to the
+  /// painted content. To use a [ColorFilter] in a [BackdropFilter], first
+  /// convert the [ColorFilter] to an [ImageFilter] using
+  /// [ImageFilter.fromColorFilter].
+  ///
+  /// A [ColorFilter] may be invalid (`ColorFilter.mode(null, null)` for
+  /// example). Attempting to use an [ImageFilter] converted from an invalid
+  /// [ColorFilter] will result in a run-time exception.
+  ///
+  /// See the [ColorFilter] class for available color filters.
+  ImageFilter.fromColorFilter(ColorFilter colorFilter)
+    : _data = Float64List(0),
+      _filterQuality = null,
+      _colorFilter = colorFilter,
+      _innerFilter = null,
+      _outerFilter = null,
+      _type = _kTypeColorFilter;
+
+  /// Composes the `inner` filter with `outer`, to combine their effects.
+  ///
+  /// Creates a single [ImageFilter] that when applied, has the same effect as
+  /// subsequently applying `inner` and `outer`, i.e.,
+  /// result = outer(inner(source)).
+  ImageFilter.compose(ImageFilter outer, ImageFilter inner)
```

Comment:
Perhaps make these named parameters so that the developer doesn't need to remember which one comes first?

The other constructors are a mixed bag of fixed and named parameters, but I think named makes good sense here.

Reply:
the parameters would need a default value if we were to make them named because of nnbd.

Reply:
Isn't there a required keyword now?